From cc5872b91123b4bd66a906bb9f46be5410669634 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Wed, 19 Feb 2025 21:35:02 +0800
Subject: [PATCH] Revert "users: Move unconfined_u definition to unconfined
 module."

This reverts commit ca3698d543c22dbc78c4c491133405754a9f8a3f.

Fix build error for targeted policy.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/kernel/kernel.te     |  3 +++
 policy/modules/system/unconfined.te | 14 --------------
 policy/users                        |  7 +++++++
 3 files changed, 10 insertions(+), 14 deletions(-)

diff --git a/policy/modules/kernel/kernel.te b/policy/modules/kernel/kernel.te
index 7ec2aa471..8f0f6ac2e 100644
--- a/policy/modules/kernel/kernel.te
+++ b/policy/modules/kernel/kernel.te
@@ -37,6 +37,9 @@ role sysadm_r;
 role staff_r;
 role user_r;
 
+# here until order dependence is fixed:
+role unconfined_r;
+
 ifdef(`enable_mls',`
 	role secadm_r;
 	role auditadm_r;
diff --git a/policy/modules/system/unconfined.te b/policy/modules/system/unconfined.te
index 71e1b15ae..940c98ce6 100644
--- a/policy/modules/system/unconfined.te
+++ b/policy/modules/system/unconfined.te
@@ -8,9 +8,6 @@ policy_module(unconfined)
 # usage in this module of types created by these
 # calls is not correct, however we dont currently
 # have another method to add access to these types
-
-role unconfined_r;
-
 userdom_base_user_template(unconfined)
 userdom_manage_home_role(unconfined_r, unconfined_t)
 userdom_manage_tmp_role(unconfined_r, unconfined_t)
@@ -273,14 +270,3 @@ unconfined_domain_noaudit(unconfined_execmem_t)
 optional_policy(`
 	unconfined_dbus_chat(unconfined_execmem_t)
 ')
-
-########################################
-#
-# Unconfined seuser
-#
-
-ifdef(`direct_sysadm_daemon',`
-        gen_user(unconfined_u, unconfined, unconfined_r system_r, s0, s0 - mls_systemhigh, mcs_allcats)
-',`
-        gen_user(unconfined_u, unconfined, unconfined_r, s0, s0 - mls_systemhigh, mcs_allcats)
-')
diff --git a/policy/users b/policy/users
index 25402afd8..ca203758c 100644
--- a/policy/users
+++ b/policy/users
@@ -28,6 +28,13 @@ gen_user(user_u, user, user_r, s0, s0)
 gen_user(staff_u, staff, staff_r sysadm_r ifdef(`enable_mls',`secadm_r auditadm_r'), s0, s0 - mls_systemhigh, mcs_allcats)
 gen_user(sysadm_u, sysadm, sysadm_r, s0, s0 - mls_systemhigh, mcs_allcats)
 
+# Until order dependence is fixed for users:
+ifdef(`direct_sysadm_daemon',`
+        gen_user(unconfined_u, unconfined, unconfined_r system_r, s0, s0 - mls_systemhigh, mcs_allcats)
+',`
+        gen_user(unconfined_u, unconfined, unconfined_r, s0, s0 - mls_systemhigh, mcs_allcats)
+')
+
 #
 # The following users correspond to Unix identities.
 # These identities are typically assigned as the user attribute
-- 
2.34.1

